File: app/src/main/java/ch/unstable/ost/database/Databases.java
package ch.unstable.ost.database;

import android.content.Context;

import androidx.room.Room;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.google.common.base.Preconditions;

import org.apache.commons.io.IOUtils;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;


public class Databases {
    static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(SupportSQLiteDatabase database) {
           database.execSQL("CREATE TABLE IF NOT EXISTS `favorite_connections` (`id` INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, `connection` TEXT NOT NULL, `creation_date` INTEGER NOT NULL)");
           database.execSQL("CREATE  INDEX `index_favorite_connections_creation_date` ON `favorite_connections` (`creation_date`)");
        }
    };
    private static final boolean FORCE_OVERRIDE = true;
    private static CacheDatabase cacheDatabase;

    public static synchronized CacheDatabase getCacheDatabase(Context context) {
        Preconditions.checkNotNull(context, "context is null");
        if (cacheDatabase == null) {
            String databaseName = "cache.db";
            cacheDatabase = Room.databaseBuilder(context.getApplicationContext(), CacheDatabase.class, databaseName)
                    .addMigrations(MIGRATION_1_2)
                    .build();
        }
        return cacheDatabase;
    }

    public static void copyDBFromAssets(Context context, String database) {
        try {
            File databaseFile = context.getApplicationContext().getDatabasePath(database);
            File parentDir = databaseFile.getParentFile();
            if (!parentDir.exists()) {
                if (!parentDir.mkdirs()) {
                    throw new IOException("Couldn't create directory: " + parentDir);
                }
            }
            if (FORCE_OVERRIDE || !databaseFile.exists()) {
                InputStream inputStream = null;
                OutputStream outputStream = null;
                try {
                    inputStream = context.getAssets().open(database);
                    outputStream = new FileOutputStream(databaseFile);
                    IOUtils.copy(inputStream, outputStream);
                } finally {
                    IOUtils.closeQuietly(inputStream);
                    IOUtils.closeQuietly(outputStream);
                }
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: app/src/main/java/ch/unstable/ost/database/Databases.kt
package ch.unstable.ost.database

import android.content.Context
import androidx.room.Room
import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase
import com.google.common.base.Preconditions
import java.io.FileOutputStream
import java.io.IOException

object Databases {
    @JvmField
    val MIGRATION_1_2: Migration = object : Migration(1, 2) {
        override fun migrate(database: SupportSQLiteDatabase) {
            database.execSQL("CREATE TABLE IF NOT EXISTS `favorite_connections` (`id` INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, `connection` TEXT NOT NULL, `creation_date` INTEGER NOT NULL)")
            database.execSQL("CREATE  INDEX `index_favorite_connections_creation_date` ON `favorite_connections` (`creation_date`)")
        }
    }
    private const val FORCE_OVERRIDE = true
    private var cacheDatabase: CacheDatabase? = null
    @JvmStatic
    @Synchronized
    fun getCacheDatabase(context: Context): CacheDatabase {
        Preconditions.checkNotNull(context, "context is null")
        var cacheDatabase = this.cacheDatabase
        if (cacheDatabase == null) {
            val databaseName = "cache.db"
            cacheDatabase = Room.databaseBuilder(context.applicationContext, CacheDatabase::class.java, databaseName)
                    .addMigrations(MIGRATION_1_2)
                    .build()
            this.cacheDatabase = cacheDatabase
            return cacheDatabase
        }
        return cacheDatabase
    }

    fun copyDBFromAssets(context: Context, database: String) {
        try {
            val databaseFile = context.applicationContext.getDatabasePath(database)
            val parentDir = databaseFile.parentFile
            if (!parentDir.exists()) {
                if (!parentDir.mkdirs()) {
                    throw IOException("Couldn't create directory: $parentDir")
                }
            }
            if (FORCE_OVERRIDE || !databaseFile.exists()) {
                context.assets.open(database).use { inputStream ->
                    FileOutputStream(databaseFile).use { outputStream ->
                        inputStream.copyTo(outputStream)
                    }
                }
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}